fastlane_bot/helpers/txhelpers.py

```diff
@@ -34,81 +31,21 @@ class TxHelpers:
-        self.web3 = self.ConfigObj.w3
-        # Set the local account
-        self.local_account = self.web3.eth.account.from_key(
-            self.ConfigObj.ETH_PRIVATE_KEY_BE_CAREFUL
-        )
-
-        # Set the public address
-        self.wallet_address = str(self.local_account.address)
-
-        self.alchemy_api_url = self.ConfigObj.RPC_URL
-        self.nonce = self.get_nonce()
-
-    def _get_transaction_info(self) -> (int, int, int, int):
-        # Get current base fee for pending block
-        current_gas_price = self.web3.eth.get_block("pending").get("baseFeePerGas")
-
-        # Get the current recommended priority fee from Alchemy, and increase it by our offset
-        current_max_priority_gas = (
-            int(
-                self.get_max_priority_fee_per_gas_alchemy()
-                * self.ConfigObj.DEFAULT_GAS_PRICE_OFFSET
-            )
-            if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"]
-            else 0
-        )
-
-        # Get current block number
-        block_number = int(self.web3.eth.get_block("latest")["number"])
-
-        # Get current nonce for our account
-        nonce = self.get_nonce()
-
-        return current_gas_price, current_max_priority_gas, block_number, nonce
-
-    def _get_prices_info(
-            self,
-            current_gas_price: int,
-            gas_estimate: int,
-            expected_profit_usd: Decimal,
-            expected_profit_eth: Decimal,
-            raw_transaction: Any
-        ) -> (int, int, int, int):
-        # Multiply expected gas by 0.8 to account for actual gas usage vs expected.
-        gas_cost_eth = (
-            Decimal(str(current_gas_price))
-            * Decimal(str(gas_estimate))
-            * Decimal(self.ConfigObj.EXPECTED_GAS_MODIFIER)
-            / Decimal("10") ** Decimal("18")
-        )
-
-        if self.ConfigObj.network.GAS_ORACLE_ADDRESS:
-            layer_one_gas_fee = self._get_layer_one_gas_fee(raw_transaction)
-            gas_cost_eth += layer_one_gas_fee
-
-        # Gas cost in usd can be estimated using the profit usd/eth rate
-        gas_cost_usd = gas_cost_eth * expected_profit_usd / expected_profit_eth
-
-        # Multiply by reward percentage, taken from the arb contract
-        adjusted_reward_eth = Decimal(Decimal(expected_profit_eth) * Decimal(self.ConfigObj.ARB_REWARD_PERCENTAGE))
-        adjusted_reward_usd = adjusted_reward_eth * expected_profit_usd / expected_profit_eth
-
-        return gas_cost_eth, gas_cost_usd, adjusted_reward_eth, adjusted_reward_usd
+        self.arb_contract = self.cfg.BANCOR_ARBITRAGE_CONTRACT
+        self.use_tenderly = self.cfg.NETWORK == self.cfg.NETWORK_TENDERLY
+        self.use_eip_1559 = self.cfg.NETWORK in ["ethereum", "coinbase_base"]
+        self.access_lists = self.cfg.NETWORK in ["ethereum"]
+        self.arb_rewards = Decimal(self.cfg.ARB_REWARDS_PPM) / 1_000_000
+        self.eth = self.cfg.w3.eth
```

Comment:
what is `self.eth`? admittedly the original sin may be in the web3 library naming convention and it is the "eth API"

https://web3py.readthedocs.io/en/stable/web3.eth.html

however -- especially for non-ETH chains this kinda does not make sense in web3, and it does not make sense here

Reply:
It is an object which exposes many of the web3 functions being used in this module.
Using it allows to slightly reduce the code length, but if you consider this a bad practice then I fix it.

Reply:
it is not bad practice it is just a bad name; what it really is is a provider, and it is not necessarily for ETH. not a killer but I'd prefer a more meaningful name

Reply:
Replaced with the original (`self.cfg.w3.eth)`.